### PR TITLE
Add Django 5 support (without Python 3.12)

### PR DESCRIPTION
**Pull request by pfouque**
Just adding last version of Django to the test matrix and package classifier

The test results are available on my repository: https://github.com/pfouque/django-cities-light/pull/1

Reworked to split with #284 (depending on the first to merge we will have to adapt)
(We will have to update the Django 5 version number anyway as soon it's officially released)

### CODE REVIEW

**Comment by marianoeramirez** [comment]
Its hard that this work. Because we need to upgrade some dependents librarys like django-dbdiff or get rid of it.

**Comment by pfouque** [comment]
Thanks for the tip @marianoeramirez, It will take time a bit more time to investigate if the best way is to find an alternative test method or update `django-dbdiff` (it has only one usage of the `imp` module but has not been updated in a while...)

**Comment by pfouque** [comment]
Depends on https://github.com/yourlabs/django-dbdiff/pull/21

File: .github/workflows/django.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8,3.9,'3.10', '3.11']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
```
Comment (adamchainz) [inline .github/workflows/django.yml:16]: Break into one-per-line to reduce future diff noise:

```suggestion
        python-version:
        - '3.8'
        - '3.9'
        - '3.10'
        - '3.11'
        - '3.12'
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,8 @@
+
```
Comment (adamchainz) [inline CHANGELOG:1]: Remove the blank line
```suggestion
```

File: tox.ini [hunk 3]
```diff
@@ -1,6 +1,10 @@
 [tox]
 envlist =
-    py{38,39,310,3.11,3.12}-django{32,40,41,42}{-sqlite,-mysql,-postgresql},
+    py{310,311,312}-django50-{sqlite,mysql,postgresql}
+    py{38,39,310,311}-django42-{sqlite,mysql,postgresql}
+    py{38,39,310,311}-django41-{sqlite,mysql,postgresql}
+    py{38,39,310}-django40-{sqlite,mysql,postgresql}
+    py{38,39,310}-django32-{sqlite,mysql,postgresql}
     checkqa,
     pylint,
```
Comment (adamchainz) [inline tox.ini:9]: You could drop these unnecessary commas whilst you're at it:
```suggestion
    checkqa
    pylint
```

File: .travis.yml [hunk 4]
```diff
@@ -5,9 +5,11 @@ services:
 cache:
```
Comment (adamchainz) [inline .travis.yml:5]: Can't this file go? I don't see any test results reported by Travis.

**Review by adamchainz — COMMENTED** [review]
Adding some minor comments.

File: tox.ini [hunk 5]
```diff
@@ -25,7 +30,8 @@ deps =
 deps =
     # sphinx
     Sphinx==4.2.0
-    django-dbdiff
+    ; django-dbdiff
+    git+https://github.com/pfouque/django-dbdiff.git@fix312#egg=django-dbdiff
```
Comment (pfouque) [inline tox.ini:34]: To be rollback after merging/releasing https://github.com/yourlabs/django-dbdiff/pull/21

```suggestion
    django-dbdiff
```

File: tox.ini [hunk 6]
```diff
@@ -36,7 +42,8 @@ deps =
     pylint
     pylint-django
     djangorestframework
-    django-dbdiff
+    ; django-dbdiff
+    git+https://github.com/pfouque/django-dbdiff.git@fix312#egg=django-dbdiff
```
Comment (pfouque) [inline tox.ini:46]: To be rollback after merging/releasing https://github.com/yourlabs/django-dbdiff/pull/21
```suggestion
    django-dbdiff
```

**Review by pfouque — COMMENTED** [review]
I temporarily added https://github.com/yourlabs/django-dbdiff/pull/21 as the source dependency and it looks fine!
https://github.com/pfouque/django-cities-light/actions/runs/6835469902

File: tox.ini [hunk 7]
```diff
@@ -57,7 +61,8 @@ deps =
     django40: Django>=4.0,<4.1
     django41: Django>=4.1,<4.2
     django42: Django>=4.2,<5.0
-    postgresql: psycopg2-binary==2.9.6
+    django50: Django==5.0rc1 # Django>=5.0,<5.1
```
Comment (pfouque) [inline tox.ini:64]:
```suggestion
    django50: Django>=5.0rc1,<5.1
```

**Comment by marianoeramirez** [comment]
@pfouque can you update your base branch with last commits?

File: setup.py [hunk 8]
```diff
@@ -46,13 +46,13 @@ def read(fname):
...
         "Framework :: Django :: 4.1",
         "Framework :: Django :: 4.2",
+        "Framework :: Django :: 5.0",
         "Programming Language :: Python",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
-        "Programming Language :: Python :: 3.12",
```
Comment (pfouque) [inline setup.py:55]: :(

**Pull request merged**